lib/elixir/lib/integer.ex

```diff
@@ -128,8 +128,13 @@ defmodule Integer do
 
   defp do_undigits([], _base, acc),
     do: acc
-  defp do_undigits([digit | tail], base, acc) when is_integer(digit),
-    do: do_undigits(tail, base, acc * base + digit)
+  defp do_undigits([digit | tail], base, acc) when is_integer(digit) do
+    if digit >= base do
+      raise ArgumentError, "invalid digit '#{digit}' in base #{base}"
```

Comment:
@lexmag sometimes we do depending on the case, e.g., https://github.com/gmile/elixir/blob/2a550c547cacba6690ca0bba13545fa824b8e867/lib/mix/lib/mix/tasks/deps.unlock.ex#L47-L48, but I agree that in this case it's easy to understand what's going on without quotes as well.